File: python3/1208.py
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        ans = 0
        i, j = 0, 0
        cnt = 0
        while i < len(s):
            cnt += abs(ord(s[i]) - ord(t[i]))
            while j < i and cnt > maxCost:
                cnt -= abs(ord(s[j]) - ord(t[j]))
                j += 1
            if i == j and cnt > maxCost:
                ans = max(ans, 0)
            else:
                ans = max(ans, i - j + 1)
            i += 1
        return ans

File: cpp/1208.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int equalSubstring(string s, string t, int maxCost) {
        int ans = 0, cnt = 0;
        int i = 0, j = 0;
        int len = s.size();
        while (i < len) {
            cnt += abs(s[i] - t[i]);
            while (j < i && cnt > maxCost) {
                cnt -= abs(s[j] - t[j]);
                ++j;
            }
            if (i == j && cnt > maxCost) {
                ans = max(ans, 0);
            } else {
                ans = max(ans, i - j + 1);
            }
            ++i;
        }
        return ans;
    }
};